utilities/calibration-delta-updater/script/blf-calibration-delta-updater.py

```diff
@@ -35,69 +33,131 @@ def open_polydriver(robot_name, control_board, joints_list):
 
-    tree = etree.parse(args.input)
-    root = tree.getroot()
-    offsets = get_offsets(root)
+    _, joints_values, _ = sensor_bridge.get_joint_positions()
 
-    control_board, sensor_bridge = open_polydriver(args.robot, args.board, args.joints)
+    # compute the offset
+    delta = joints_values[joint_index] * 180 / math.pi - expected_values[joint_index]
+    return delta
 
-    print_info("Please put the joints in zero configuration.")
-    print_info("Press enter and I will show you the offsets.")
-    input()
 
-    assert sensor_bridge.advance() is True
+def parse_handler(file, robot_part):
+    # load the parameter handler
+    parameters = blf.parameters_handler.YarpParametersHandler()
 
-    _, joints_values, _ = sensor_bridge.get_joint_positions()
-    new_offsets = np.array(offsets) + joints_values * 180 / math.pi
+    print("---------------------------------->" + file)
+
+    if os.path.isfile(file):
+        if not parameters.set_from_file_path(file):
+            raise RuntimeError('Unable to load the configuration file')
+    else:
+        if not parameters.set_from_filename(file):
+            raise RuntimeError('Unable to load the configuration file')
 
-    print_info("Joint values = " + str(['%.4f' % joint for joint in joints_values * 180 / math.pi]) + " deg")
-    print_info("previous offset = " + str(['%.4f' % offset for offset in offsets]) + " deg")
-    print_info("new offset = " + str(['%.4f' % offset for offset in new_offsets]) + " deg")
+    group = parameters.get_group(robot_part)
+    if group is None:
+        raise RuntimeError('Unable to find the group named ' + robot_part)
 
+    joints = group.get_parameter_vector_string('joints_list')
+    board = group.get_parameter_string('control_board')
+    parameters.set_parameter_vector_string('joints_list', joints)
+    parameters.set_parameter_vector_string('remote_control_boards', [board])
+    values = group.get_parameter_vector_float('expected_values')
+
+    # populate the parameters handler with additional parameters required by the application. There is no need to
+    # expose these parameter in the configuration file
+    parameters.set_parameter_string('local_prefix', 'joint-offset-updater')
+    parameters.set_parameter_bool('check_for_nan', False)
+    parameters.set_parameter_bool('stream_joint_states', True)
+    return parameters, values, joints
+
+
+def main():
+    parser = argparse.ArgumentParser(description='Automatically update the calibration delta of the robot '
+                                                 'configuration file.')
+    parser.add_argument('-i', '--input', type=str, required=True, help='Input xml file')
+    parser.add_argument('-o', '--output', type=str, required=True, help='Output xml file')
+    parser.add_argument('-p', '--part', type=str, required=True, help='Name of the part.')
+    parser.add_argument('--config', type=str, required=False, help='Path or name of the configuration file',
```

Comment:
This is slightly different from what is mentioned in the README. Not a big deal, but it could raise doubts